libraries/aws/shadow/src/aws_iot_shadow_operation.c

```diff
@@ -272,115 +269,107 @@ static void _commonOperationCallback( _shadowOperationType_t type,
     if( AwsIot_ParseThingName( pMessage->u.message.info.pTopicName,
                                pMessage->u.message.info.topicNameLength,
                                &( param.pThingName ),
-                               &( param.thingNameLength ) ) == false )
+                               &( param.thingNameLength ) ) == true )
     {
-        IOT_GOTO_CLEANUP();
-    }
-
-    /* Lock the pending operations list for exclusive access. */
-    IotMutex_Lock( &( _AwsIotShadowPendingOperationsMutex ) );
-
-    /* Search for a matching pending operation. */
-    pOperationLink = IotListDouble_FindFirstMatch( &( _AwsIotShadowPendingOperations ),
-                                                   NULL,
-                                                   _shadowOperation_match,
-                                                   &param );
-
-    /* Find and remove the first Shadow operation of the given type. */
-    if( pOperationLink == NULL )
-    {
-        /* Operation is not pending. It may have already been processed. Return
-         * without doing anything */
-        IotMutex_Unlock( &( _AwsIotShadowPendingOperationsMutex ) );
-
-        IotLogWarn( "Shadow %s callback received an unknown operation.",
-                    _pAwsIotShadowOperationNames[ type ] );
+        /* Lock the pending operations list for exclusive access. */
+        IotMutex_Lock( &( _AwsIotShadowPendingOperationsMutex ) );
 
-        IOT_GOTO_CLEANUP();
-    }
-    else
-    {
-        pOperation = IotLink_Container( _shadowOperation_t, pOperationLink, link );
+        /* Search for a matching pending operation. */
+        pOperationLink = IotListDouble_FindFirstMatch( &( _AwsIotShadowPendingOperations ),
+                                                       NULL,
+                                                       _shadowOperation_match,
```

Comment:
(Minor) Not sure if this function should be named as `_shadowOperationMatch` to follow the naming convention

Reply:
Agree. Will fix.